cvmfs/kvstore.h

```diff
@@ -0,0 +1,208 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_KVSTORE_H_
+#define CVMFS_KVSTORE_H_
+
+#include <pthread.h>
+#include <unistd.h>
+
+#include <string>
+
+#include "cache.h"
+#include "lru.h"
+#include "statistics.h"
+
+using namespace std;  // NOLINT
+
+namespace kvstore {
+
+struct MemoryBuffer {
+  void *address;
+  size_t size;
+  unsigned int refcount;
+  cache::CacheManager::ObjectType object_type;
```

Comment:
There are more object types defined than kvstores, so I included the field to avoid dropping information. We would also need it for e.g. determining if a pinned entry has any actual references (refcount starts at 1 instead of 0). I don't really make use of it, so if the rest of the code doesn't care about object types, I could remove it.